build-rules/build.make

```diff
@@ -0,0 +1,103 @@
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The following make rules support building one or more command,
+# packaging them in a containers and pushing to a registry. The expected
+# repository layout is:
+# - cmd/*/*.go - source code for each command
+# - cmd/*/Dockerfile - docker file for each command
```

Comment:
Actually, do we have any repos where there are multiple binaries being built?  If not, can we keep the Dockerfile at the root?  Moving the Dockerfile will break existing CI and make it more complicated to manage the build process across release branches.

Reply:
I don't think we do. "drivers" would be the only candidate, but even that only ever published the hostpath driver.

We can support both: use `cmd/*/Dockerfile` if it exists, otherwise `Dockerfile` in the root.